### PR TITLE
applied sorting to posts and shares, optimized the controller

### PR DESCRIPTION
**Pull request by jeromes123**
**OVERVIEW**
Fixed the display of posts and shares where all shares are displayed after the original posts.  The original posts and shared posts are now sorted by their updated_at attribute.

### CODE REVIEW

File: src/app/Services/UserPostService.php [hunk 1]
```diff
@@ -96,8 +104,17 @@ public function getAllPostsAndMedia(): array
...
+        
+        $postsMediaAndShares = $postsMediaAndShares->sortBy(function ($item) {
+            if (is_array($item)) {
+                return $item['post']->updated_at;
+            } elseif ($item instanceof \App\Models\PostShare) {
+                return $item->updated_at;
+            }
+        });
+
+        return $postsMediaAndShares ;
```
Comment (mikco-hipe) [inline src/app/Services/UserPostService.php]: Can you try this?
```suggestion
$postsMediaAndShares = $postsMediaAndShares->sortBy(function ($item) {
    if ($item instanceof \App\Models\PostShare) {
        return $item->updated_at;
    }

    return $item['post']->updated_at;
});
```

**Review by mikco-hipe — COMMENTED** [review]
Please check!

File: src/app/Services/UserPostService.php (hunk 1 above)
Comment (jeromes123) [reply src/app/Services/UserPostService.php]: Ill try sir

File: src/app/Services/UserPostService.php [hunk 2]
```diff
@@ -81,9 +89,12 @@ public function getUserPostAndMediaToEdit(int $post_id): Array
     }
 
     /**
-     * Gets all the posts and media from database
+     * Gets all the user posts, posts media
+     * Combine the all posts and their media into array<UserPost, PostMedia>
+     * Combine array<UserPost, PostMedia> with all PostShares 
+     * Sort the Collection of array<UserPost, PostMedia> and PostShares by date
```
Comment (mikco-hipe) [inline src/app/Services/UserPostService.php]: Can you lessen the comments?

Comment (jeromes123) [reply src/app/Services/UserPostService.php]: will do sir

**Review by hipe01mark — APPROVED** [review]
LGTM 🙇

**Pull request merged**